docs/litecoin.md

```diff
@@ -1,43 +1,99 @@
+{
+  "mhknqLHQGWDXuLsPdzab8nA4jD3fMdVYS2": {
+    "purpose": "receive"
+  },
+  "myGi4qfsqJ8kzgLEJLfdqr75PQQFyH2DSf": {
+    "purpose": "receive"
+  },
+  "QjpdvL4h5jnfaj1uV5ifJNUAYZTTbjgFH5": {
+    "purpose": "receive"
+  },
+  "tltc1qp4t37xntgjw43lsuc3xr40hg2akne6j6twy3xn": {
+    "purpose": "receive"
+  },
+  "tltc1qrz8z67vtu38pq2yzqtq7unftmsaueq6a8da5n2": {
+    "purpose": "receive"
+  },
+  "tltc1qyth2j9ar6gzkuhuu7qytl9x8q2ud5p7fjts2vy": {
+    "purpose": "receive"
+  },
+  "tltc1qjadecaemcwpk8ytjclns56vq2kurd7f3j3dcx7": {
+    "purpose": "receive"
+  },
+  "tmweb1qqd6usenz75lvym9px3q7gakhy4a3u4ne95rwg4h3w2shm7hva28wsqemjfm7c2v7gt5sl5snf9kr6tygl3t773l6spt4cmuel4d92m038gtejwnn": {
+    "purpose": "receive"
+  }
+}
+```
+
+The legacy addresses are normally with short length (see [How is a Litecoin address generated?](https://bitcoin.stackexchange.com/questions/65282/how-is-a-litecoin-address-generated) for details).
+Here both the short addresses `mhknqLHQGWDXuLsPdzab8nA4jD3fMdVYS2` and `myGi4qfsqJ8kzgLEJLfdqr75PQQFyH2DSf` may
+be used to sign message, while another short address `QjpdvL4h5jnfaj1uV5ifJNUAYZTTbjgFH5` does not work.
+Which address works and which does not work may require some trial and error.
+
+If the account is not labelled, you may run `litecoin-cli -testnet listreceivedbyaddress 1 true`
+to dump all addresses and find out the desired legacy address from there.
+
+## Obtaining public key hash needed by ckb-auth
+To obtain public key hash of address `mhknqLHQGWDXuLsPdzab8nA4jD3fMdVYS2`, we need to decode it with
+base58 and then take the [1, 21) bytes of the resulting binary data (i.e. dropping the first byte, and
+take the following 20 bytes of data). You may [decode it online](http://lenschulwitz.com/base58) or run
+the command `base58 -d <<< mhknqLHQGWDXuLsPdzab8nA4jD3fMdVYS2 | xxd -s 1 -l 20 -p` 
+([keis/base58](https://github.com/keis/base58) and [xxd(1)](https://linux.die.net/man/1/xxd)required).
 
-### Signing the message
-Signing is easier after obtained the wallet address and imported the private key. Running
-`SIGNATURE="$(litecoin-cli -rpcwallet=ckb-auth-test-wallet -testnet signmessage "$ADDRESS" "$MESSAGE")"` is enough.
-You can also verify the result with `litecoin-cli -rpcwallet=ckb-auth-test-wallet -testnet verifymessage "$ADDRESS" "$SIGNATURE" "$MESSAGE"`.
+## Signing a message with litecoin-cli
+To sign the message `29553f9e37fa16e45f1d3e616ac5366f6afd9936477f2d6fc870f49bdf540157`, we can run
+`litecoin-cli -rpcwallet=ckb-auth-test-wallet -testnet signmessage mhknqLHQGWDXuLsPdzab8nA4jD3fMdVYS2 29553f9e37fa16e45f1d3e616ac5366f6afd9936477f2d6fc870f49bdf540157`,
+whose output `H2gWHSzffPkvD2QYD9jdLTuM5x7cax4E7Ax0Nm79T90wLU0LcK/DEYENggsC5G4FK/69qJP9eXp9zHkZ40PH6ME=` is the base64-encoded form of the signature. The binary form can be fed to ckb-auth for authentication.
 
```

Comment:
Should we add how to use `ckb-auth` for verification in the contract into the document?

Reply:
This is not specific to litecoin. We should use this document [ckb-auth/auth.md at main · cryptape/ckb-auth · GitHub](https://github.com/cryptape/ckb-auth/blob/main/docs/auth.md). I added a link to that.